File: NameSiloDynDns/HttpClientSetup/HttpClientExtensions.cs
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.DependencyInjection.Extensions;
using Microsoft.Extensions.Http;

namespace NameSiloDynDns.HttpClientSetup
{
    public static class HttpClientExtensions
    {
        public static IHttpClientBuilder ConfigureNameSiloHttpLogging(this IHttpClientBuilder builder)
        {
            builder.Services.Replace(ServiceDescriptor.Singleton<IHttpMessageHandlerBuilderFilter, NameSiloHttpClientFilter>());

            return builder;
        }

    }
}

File: NameSiloDynDns/HttpClientSetup/HttpClientExtensions.fs
[<AutoOpen>]
module NameSiloDynDns.HttpClientSetup.Extensions

open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.DependencyInjection.Extensions
open Microsoft.Extensions.Http

type IHttpClientBuilder with
    member builder.ConfigureNameSiloHttpLogging() =
        builder.Services.Replace(
            ServiceDescriptor.Singleton<IHttpMessageHandlerBuilderFilter, NameSiloHttpClientFilter>()
        )
        |> ignore

        builder
